G/GDAL/build_tarballs.jl

```diff
@@ -13,74 +13,42 @@ version = VersionNumber(upstream_version.major * 100 + version_offset.major,
 sources = [
     ArchiveSource("https://github.com/OSGeo/gdal/releases/download/v$upstream_version/gdal-$upstream_version.tar.gz",
         "7c4406ca010dc8632703a0a326f39e9db25d9f1f6ebaaeca64a963e3fac123d1"),
-    DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/gdal-*/
+mkdir build
+cd build
 
-if [[ ${target} == *mingw* ]]; then
-    export LDFLAGS="-L${libdir}"
-    # Apply patch to customise PROJ library
-    atomic_patch -p1 "$WORKSPACE/srcdir/patches/configure_ac_proj_libs.patch"
-    autoreconf -vi
-    export PROJ_LIBS="proj_9_1"
-elif [[ "${target}" == *-linux-* ]]; then
-    # Hint to find libstdc++, required to link against C++ libs when using C compiler
-    if [[ "${nbits}" == 32 ]]; then
-        export CFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/lib"
-    else
-        export CFLAGS="-Wl,-rpath-link,/opt/${target}/${target}/lib64"
-    fi
-    # Use same flags also for GEOS
-    atomic_patch -p1 "$WORKSPACE/srcdir/patches/geos-m4-extra-cflags.patch"
-    export EXTRA_GEOS_CFLAGS="${CFLAGS}"
-    if [[ "${target}" == powerpc64le-* ]]; then
-        atomic_patch -p1 "$WORKSPACE/srcdir/patches/sqlite3-m4-extra-libs.patch"
-        export EXTRA_GEOS_LIBS="${EXTRA_GEOS_LIBS} -lm"
-        export EXTRA_SQLITE3_LIBS="-lm"
-        # libpthread and libldl are needed for libgdal, so let's always use them
-        export LDFLAGS="$LDFLAGS -lpthread -ldl"
-    fi
-    autoreconf -vi
+if [[ "${target}" == *-freebsd* ]]; then
+    # Our FreeBSD libc has `environ` as undefined symbol, so the linker will
+    # complain if this symbol is used in the built library, even if this won't
+    # be a problem at runtime. This flag allows having undefined symbols.
+    export LDFLAGS="-undefined"
```

Comment:
You're shifting a linking time error to a runtime error, it doesn't look like a great tradeoff

Reply:
Ugh, I see, it's https://docs.binarybuilder.org/stable/troubleshooting/#undefined-reference-to-environ'-1.  Sigh

Reply:
Yeah it's used in a `posix_spawnp` call. Interestingly that file got this FreeBSD patch recently: https://github.com/OSGeo/gdal/commit/4ff30d57c654f34d04eb1cc5b2b76e4d52c78b35

Reply:
Then it'd probably be better to apply that patch than to change link flags.

Reply:
Sorry, that patch is already included. With or without it I get the same issue, I tried playing around with it a bit. In https://bugs.webkit.org/show_bug.cgi?id=138420 they resolved it by avoiding using `environ` altogether. Don't know if that is feasible. It is only used once, in this function:

https://github.com/OSGeo/gdal/blob/v3.5.1/port/cpl_spawn.cpp#L578-L612

On this line:

https://github.com/OSGeo/gdal/blob/v3.5.1/port/cpl_spawn.cpp#L708